### PR TITLE
Chk 2992 shared api key pdv

### PR DESCRIPTION
**Pull request by CianoDanilo**
#### List of Changes

This feature will add a decrypt with a different api key for user's fiscal code
#### How Has This Been Tested?
This feature has been tested with JUnit tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: helm/values-dev.yaml [hunk 1]
```diff
@@ -181,7 +181,8 @@ microservice-chart:
     QUEUE_DEADLETTER_CONNECTION_STRING: ecommerce-storage-deadletter-connection-string
     NOTIFICATIONS_SERVICE_API_KEY: notifications-service-api-key
     TRANSACTIONS_SERVICE_API_KEY: transactions-service-auth-update-api-key
-    PERSONAL_DATA_VAULT_API_KEY: personal-data-vault-api-key
+    ECOMMERCE_PERSONAL_DATA_VAULT_API_KEY: personal-data-vault-api-key
+    SHARED_PERSONAL_DATA_VAULT_API_KEY: personal-data-vault-api-key-wallet-session
```
Comment (giovanniberti) [inline helm/values-dev.yaml:185]: _(Placeholder comment for key creation on key vaults)_

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/config/SecretConfigurations.kt [hunk 2]
```diff
@@ -3,13 +3,14 @@ package it.pagopa.ecommerce.eventdispatcher.config
...
 import it.pagopa.generated.pdv.v1.api.TokenApi
+import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 
 @Configuration
 class SecretConfigurations {
-  @Bean
+  @Bean("personalDataVaultApi")
```
Comment (giovanniberti) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/config/SecretConfigurations.kt:13]: We could call this `eCommercePersonalDataVaultApi` for consistency, what do you think?

```suggestion
  @Bean("eCommercePersonalDataVaultApi")
```

**Review by giovanniberti — DISMISSED** [review]
🚀

File: .env.example [hunk 3]
```diff
@@ -55,7 +55,8 @@ CLOSE_PAYMENT_RETRY_EVENT_BASE_INTERVAL_SECONDS=60
 CLOSE_PAYMENT_RETRY_EVENT_MAX_ATTEMPTS=3
 NOTIFICATION_RETRY_EVENT_BASE_INTERVAL_SECONDS=600
 NOTIFICATION_RETRY_EVENT_MAX_ATTEMPTS=3
-PERSONAL_DATA_VAULT_API_KEY=personal-data-vault-api-key
+ECOMMERCE_PERSONAL_DATA_VAULT_API_KEY=personal-data-vault-api-key
+SHARED_PERSONAL_DATA_VAULT_API_KEY=personal-data-vault-api-key-wallet-session
```
Comment (infantesimone) [inline .env.example:59]: proposal, why not rename everything to `wallet-session`? even if we refer to `shared`, it is  the domain to which the wallet-session-pagopa component belongs

Comment (CianoDanilo) [reply .env.example:59]: I agree, i'm going to rename it

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/config/SecretConfigurations.kt (hunk 2 above)
Comment (CianoDanilo) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/config/SecretConfigurations.kt:13]: I agree, I'm going to rename it

File: .env.example (hunk 3 above)
Comment (infantesimone) [reply .env.example:59]: or PAGOPA_SESSION :-)

Comment (infantesimone) [reply .env.example:59]: pr PAGOPA_PLATFORM_SESSION :-)

Comment (infantesimone) [reply .env.example:59]: I correct myself :-)  better `WALLET_SESSION_` which reflects the name of the secret in KV (and we leave it equal to the name defined in the SHARED domain).
what do you think about it? @giovanniberti @CianoDanilo

Comment (CianoDanilo) [reply .env.example]: Changed prefix to WALLET_SESSION

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**